Organize utils

jaredthomas68 opened the pull request:
This PR reorganizes the utils and related tests to allow for more manageable extension of the utilities.

cfrontin reviewed (changes requested):
Overall this looks great; two comments to deal with:
- are we making the FLORIS config generation a core feature on the util level or does it belong more on the level of a ride-along function on the aero component level?
- can you add either a readme or file-header comments to the utility functions to lay out the philosophy of the utils? maybe a list of things that can be found in each module, too to help guide future generation of util functionality

jaredthomas68 commented:
> Overall this looks great; two comments to deal with:
> 
> * are we making the FLORIS config generation a core feature on the util level or does it belong more on the level of a ride-along function on the aero component level?
> * can you add either a readme or file-header comments to the utility functions to lay out the philosophy of the utils? maybe a list of things that can be found in each module, too to help guide future generation of util functionality

1. I went ahead and moved it to Floris.py and moved `load_turbine_spec` to a new file (`utils/io.py`) for lack of a better place to put it right now. I think it is definitely time to have some more API planning discussions
2. I added a page on the docs that discusses the utils philosophy. I think having it in the docs is more extensible.

cfrontin reviewed (approved):
Looks great!
